Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -108,6 +117,34 @@ var JSTimersExecution = {
     }
   },
 
+  callIdleCallbacks: function(frameTime) {
+    // Arbitrary threshold: 10ms remaining in frame.
+    if (17 - (performanceNow() - frameTime) < 10) {
```

Comment:
The only key here is that this should be *longer* than a frame. It should be possible to go beyond 17ms in an idle callback while still hitting every frame deadline. E.g. 5ms to process the current frame, 20ms to process the idle callback, 5ms to process the next frame. Still hits both deadlines.

If you don't call it often enough, then you're leaving CPU cycles unused. Only calling this if there is 10ms left is way too restrictive because there is too much headroom left out. Remember, even if it goes beyond the frame deadline, it doesn't matter. This should be like 1ms or even less.